### ISSUE TITLE
Allow dynamic route segments in @astrojs/vercel isr.exclude array

### ISSUE THREAD

**Issue by tk04** [withastro/astro#10492]
### Astro Info

```block
Astro                    v4.5.6
Node                     v20.11.0
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/vercel/serverless
Integrations             none
```

### Describe the Bug

When using the @astrojs/vercel integration, I want to exclude a dynamic route from using ISR. Currently, there seems to be no way to do this. All the routes that are passed into the `isr.exclude` array are matched exactly as is.

https://github.com/withastro/astro/blob/ad57a02c330b544770ab853fe0521eb784421016/packages/integrations/vercel/src/serverless/adapter.ts#L354-L357

### What's the expected result?

The processing of the `isr.exclude` array should offer a way to match dynamic route segments. Therefore, when the code is built, the route `src` values in the`.vercel/output/config.json` file should reflect dynamic route segments accordantly.

The strings are currently processed like this:
https://github.com/withastro/astro/blob/ad57a02c330b544770ab853fe0521eb784421016/packages/integrations/vercel/src/lib/redirects.ts#L79-L81

It would be nice to enable dynamic values using `/:path` or `/[path]`. A solution might look something like this:
```typescript
function escapeRegex(content) {
  return `^${getMatchPattern(
    content.split("/").map((value) => [
      {
        content: value,
        dynamic: value.startsWith(":"),
        spread: false,
      },
    ]),
  )}$`;
}


```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-alhqnq?file=astro.config.mjs

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
This sounds more like a feature request, not a bug report. 

As far as I remember, ISR accepts regex too.

**Comment by tk04** [comment]
The route `src` values in the `config.json` file allow regex, but the `exclude.isr` array doesn't, the values are always escaped:

https://github.com/withastro/astro/blob/ad57a02c330b544770ab853fe0521eb784421016/packages/integrations/vercel/src/serverless/adapter.ts#L354-L357

**Comment by lilnasy** [comment]
The exclude field currently only accepts verbatim paths, but I think it makes sense for it to handle dynamic and spread segments like `/[path]`.

It would work similarly to how configured redirects are processed. We have utilities somewhere that turn `/api/[path]` into the regular expression `/api/([^/]+?)`.

**Comment by lilnasy** [comment]
This would be a new feature.

**Comment by lilnasy** [comment]
Implemented in #10513. Thanks @tk04!
